File: 윤석주/day10/행복한지 슬픈지.py
message = input()
#감정 갯수
sad = 0
happy = 0

#:가 나온 경우 다음, 다다음의 인덱스 여부를 확인한 뒤 알맞은 감정 +1
for i in range(len(message)):
    if message[i] == ':':
        if i+1<len(message) and message[i+1] == '-':
            if i+2<len(message):
                if message[i+2] == ')':
                    happy+=1
                elif message[i+2] == '(':
                    sad+=1

#감정 갯수별 프린트
if not sad and not happy:
    print("none")
elif happy > sad:
    print("happy")
elif happy < sad:
    print("sad")
else:
    print("unsure")

File: 박재현/이전/행복한지슬픈지.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int main(void)
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    string str;
    int happy = 0;
    int sad = 0;
    getline(cin, str);

    int size = str.length();

    if (size < 3)
    {
        cout << "none";
        return 0;
    }

    for (int i = 0; i <= size - 2; i++)
    {
        if (str.substr(i, 3) == ":-)")
        {
            happy++;
            i += 2;
        }
        else if (str.substr(i, 3) == ":-(")
        {
            sad++;
            i += 2;
        }
    }

    if (sad > happy)
    {
        cout << "sad";
    }
    else if (sad < happy)
    {
        cout << "happy";
    }
    else if (sad == happy)
    {
        cout << "unsure";
    }
    else
    {
        cout << "none";
    }

    return 0;
}